File: Continuous Probability Distributions/R/UniformeContinua/UniformeContinua.R
# Versão Distribuição Uniforme Contínua
# Carregar as bibliotecas necessárias
if(!require(ggplot2)) install.packages("ggplot2"); library(ggplot2)
# Para juntar os gráficos lado a lado
if(!require(patchwork)) install.packages("patchwork"); library(patchwork)

# 1. Definição dos parâmetros da Distribuição Uniforme Contínua
a <- 2   # Limite inferior do suporte
b <- 8   # Limite superior do suporte

# Definir uma margem visual para mostrar a densidade antes de 'a' e depois de 'b'
margem <- (b - a) * 0.2
x_min  <- a - margem
x_max  <- b + margem

# 2. Suporte: Valores contínuos incluindo a margem externa
x <- seq(x_min, x_max, length.out = 1000)

# 3. Cálculo das funções exatas e criação do data.frame
dados_uniforme_cont <- data.frame(
  x  = x,
  fx = dunif(x, min = a, max = b), # PDF: Densidade de probabilidade (Retangular)
  Fx = punif(x, min = a, max = b)  # CDF: Probabilidade acumulada (Rampa linear)
)

# ==============================================================================
# Gráfico da PDF (Função de Densidade de Probabilidade)
# ==============================================================================
g1 <- ggplot(dados_uniforme_cont, aes(x = x, y = fx)) +
  # Desenha a linha contínua que forma o retângulo da densidade uniforme
  geom_line(color = "steelblue", linewidth = 1) +
  # Preenche a área sob o retângulo para melhor visualização
  geom_area(fill = "steelblue", alpha = 0.15) +
  # Configurações de eixos e títulos dinâmicos
  scale_x_continuous(expand = c(0, 0), breaks = seq(round(x_min), round(x_max), by = 1)) +
  scale_y_continuous(limits = c(0, (1 / (b - a)) * 1.3), expand = c(0, 0)) +
  labs(
    title = paste0("Função de Densidade (PDF) - U(", a, ", ", b, ")"),
    x = "Valor (x)",
    y = "f(x)"
  ) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))

# ==============================================================================
# Gráfico da CDF (Função de Distribuição Acumulada Contínua)
# ==============================================================================
g2 <- ggplot(dados_uniforme_cont, aes(x = x, y = Fx)) +
  # Desenha a linha contínua (0 antes de 'a', sobe em linha reta, estabiliza em 1 após 'b')
  geom_line(color = "darkorange", linewidth = 1) +
  # Configurações de eixos e títulos
  scale_x_continuous(expand = c(0, 0), breaks = seq(round(x_min), round(x_max), by = 1)) +
  scale_y_continuous(limits = c(0, 1.05), breaks = seq(0, 1, 0.2), expand = c(0, 0)) +
  labs(
    title = "Distribuição Acumulada (CDF)",
    x = "Valor (x)",
    y = "F(x) = P(X <= x)"
  ) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))

# ==============================================================================
# Exibir e salvar os gráficos lado a lado
# ==============================================================================
 grafico_final <- g1 + g2
 x11()
 grafico_final
# Salvar como JPEG
# ggsave("figUniformeContinua.jpeg", plot = grafico_final, width = 10, height = 6, dpi = 300)

File: Continuous Probability Distributions/Python/UniformeContinua/UniformeContinua.py
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import uniform

# Versão Distribuição Uniforme Contínua

# 1. Definição dos parâmetros da Distribuição Uniforme Contínua
a = 2   # Limite inferior do suporte
b = 8   # Limite superior do suporte

# O SciPy usa a parametrização: loc = mínimo, scale = largura (b - a)
loc_param = a
scale_param = b - a

# Definir uma margem visual para mostrar a densidade antes de 'a' e depois de 'b'
margem = (b - a) * 0.2
x_min = a - margem
x_max = b + margem

# 2. Suporte: Valores contínuos incluindo a margem externa
x = np.linspace(x_min, x_max, 1000)

# 3. Cálculo das funções exatas
# uniform.pdf equivale a dunif (Densidade)
# uniform.cdf equivale a punif (Acumulada)
fx = uniform.pdf(x, loc=loc_param, scale=scale_param)
Fx = uniform.cdf(x, loc=loc_param, scale=scale_param)

# ==============================================================================
# Configuração da figura para os gráficos lado a lado (equivalente ao patchwork)
# ==============================================================================
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 5))

# Estilo visual limpo (equivalente ao theme_classic)
for ax in [ax1, ax2]:
    ax.spines['top'].set_visible(False)
    ax.spines['right'].set_visible(False)
    ax.tick_params(top=False, right=False)

# Criando a sequência de marcas nos eixos (breaks) pulando de 1 em 1
marcas_eixo_x = np.arange(int(round(x_min)), int(round(x_max)) + 1, 1)

# ------------------------------------------------------------------------------
# Gráfico 1: PDF (Função de Densidade de Probabilidade)
# ------------------------------------------------------------------------------
# Desenha a linha contínua que forma o retângulo da densidade uniforme
ax1.plot(x, fx, color="steelblue", linewidth=2)
# Preenche a área sob o retângulo para melhor visualização (alpha controla a transparência)
ax1.fill_between(x, fx, color="steelblue", alpha=0.15)

# Configurações de eixos e títulos dinâmicos
ax1.set_xlim(x_min, x_max)
ax1.set_xticks(marcas_eixo_x)
ax1.set_ylim(0, (1 / (b - a)) * 1.3)
ax1.set_title(f"Função de Densidade (PDF) - U({a}, {b})", weight='bold', pad=15)
ax1.set_xlabel("Valor (x)")
ax1.set_ylabel("f(x)")

# ------------------------------------------------------------------------------
# Gráfico 2: CDF (Função de Distribuição Acumulada Contínua)
# ------------------------------------------------------------------------------
# Desenha a linha contínua (0 antes de 'a', sobe em linha reta, estabiliza em 1 após 'b')
ax2.plot(x, Fx, color="darkorange", linewidth=2)

# Configurações de eixos e títulos
ax2.set_xlim(x_min, x_max)
ax2.set_xticks(marcas_eixo_x)
ax2.set_ylim(0, 1.05)
ax2.set_yticks(np.arange(0, 1.1, 0.2))  # Quebras de 0 até 1 de 0.2 em 0.2
ax2.set_title("Distribuição Acumulada (CDF)", weight='bold', pad=15)
ax2.set_xlabel("Valor (x)")
ax2.set_ylabel("F(x) = P(X <= x)")

# Ajusta o espaçamento para não cortar os textos
plt.tight_layout()

# Salvar como JPEG (Equivalente ao ggsave)
# plt.savefig("figUniformeContinua.jpeg", dpi=300, format="jpeg")

# Exibir os gráficos na tela (Equivalente ao x11() + plot)
plt.show()
